schematic/utils/general.py

```diff
@@ -57,24 +60,66 @@ def get_dir_size(path: str):
                 total += get_dir_size(entry.path)
     return total
 
+def calculate_datetime(minutes: int, input_date: datetime, before_or_after: str = "before") -> datetime:
```

Comment:
One thought to simplify this function is to allow for negative or positive minutes. Not sure if that's allowed in the timedelta though.